langchain4j-mcp/src/main/java/dev/langchain4j/mcp/client/transport/stdin/StdioMcpTransport.java

```diff
@@ -0,0 +1,132 @@
+package dev.langchain4j.mcp.client.transport.stdin;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import dev.langchain4j.mcp.client.protocol.McpCallToolRequest;
+import dev.langchain4j.mcp.client.protocol.McpInitializeRequest;
+import dev.langchain4j.mcp.client.protocol.McpListToolsRequest;
+import dev.langchain4j.mcp.client.transport.McpTransport;
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ConcurrentHashMap;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class StdioMcpTransport implements McpTransport {
```

Comment:
Is there a test for ToolSpecification? Would it make sense to introduce this class along with the test in a separate PR to keep PRs small?